kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/TopicEncryptionST.java

```diff
@@ -0,0 +1,122 @@
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+import io.kroxylicious.systemtests.utils.NamespaceUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@ExtendWith(KroxyliciousExtension.class)
+class TopicEncryptionST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(TopicEncryptionST.class);
+    private static final String MESSAGE = "Hello-world";
+    private static KubeVaultTestKmsFacade kubeVaultTestKmsFacade;
+    private final String clusterName = "my-cluster";
+    private String bootstrap;
+
+    @BeforeAll
+    void setUp() {
+        kubeVaultTestKmsFacade = new KubeVaultTestKmsFacadeFactory().build(Vault.VAULT_DEFAULT_NAMESPACE, Vault.VAULT_POD_NAME);
+        List<Pod> vaultPods = kubeClient().listPodsByPrefixInName(Vault.VAULT_DEFAULT_NAMESPACE, Vault.VAULT_SERVICE_NAME);
+        if (!vaultPods.isEmpty()) {
+            LOGGER.warn("Skipping vault deployment. It is already deployed!");
+        }
+        else {
+            NamespaceUtils.createNamespaceWithWait(Vault.VAULT_DEFAULT_NAMESPACE);
+            kubeVaultTestKmsFacade.start();
+        }
+
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KROXY_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.warn("Skipping kafka deployment. It is already deployed!");
+            return;
+        }
+        LOGGER.info("Deploying Kafka in {} namespace", Constants.KROXY_DEFAULT_NAMESPACE);
+
+        Kafka kafka = KafkaTemplates.kafkaPersistentWithKRaftAnnotations(Constants.KROXY_DEFAULT_NAMESPACE, clusterName, 3).build();
+
+        resourceManager.createResourceWithWait(
+                KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka, 3).build(),
+                kafka);
+    }
+
+    @BeforeEach
+    void beforeEach(String namespace) {
+        var testKekManager = kubeVaultTestKmsFacade.getTestKekManager();
+        testKekManager.generateKek(topicName);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        Kroxylicious kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithTopicEncryptionFilter(clusterName, 1, topicName);
+        bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And KafkaTopic in {} namespace", namespace);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+    }
+
+    @AfterAll
+    void cleanUp() {
+        if (kubeVaultTestKmsFacade != null) {
+            kubeVaultTestKmsFacade.stop();
+        }
+    }
+
+    @Test
+    void produceAndConsumeEncryptedMessage(String namespace) {
```

Comment:
Both tests are about encrypted messages, so the reader won't understand the difference between the two tests.  Let's give this one a better name.

How about?
`ensureClusterHasEncryptedMessage`

Reply:
great! I'm horrible giving names...